lib/main.ts

```diff
@@ -166,80 +170,99 @@ module compile {
 		// Unsupported by gulp-typescript
 		sourceRoot?: string; // Use sourceRoot in gulp-sourcemaps instead
 	}
+
+	export interface GulpTsSettings {
+		useFileCompiler?: boolean;
+		typescript?: typeof ts;
+	}
+
 	export type Project = _project.Project;
 	export type CompileStream = _project.ICompileStream;
 	export import reporter = _reporter;
 
-	export function createProject(tsConfigFileName: string, settings?: Settings): Project;
-	export function createProject(settings?: Settings): Project;
-	export function createProject(fileNameOrSettings?: string | Settings, settings?: Settings): Project {
+	function _createProject({
```

Comment:
Normal arguments instead of an object, as you are giving all properties at both calls. So an object does not really add a value here.